apps/prairielearn/src/lib/course-request.ts

```diff
@@ -0,0 +1,104 @@
+import { loadSqlEquiv, queryRow, queryAsync } from '@prairielearn/postgres';
+import { z } from 'zod';
+import { logger } from '@prairielearn/logger';
+import * as Sentry from '@prairielearn/sentry';
+
+import { DateFromISOString, IdSchema } from '../lib/db-types';
+import { createCourseRepoJob } from '../lib/github';
+import { sendCourseRequestMessage } from '../lib/opsbot';
+
+const sql = loadSqlEquiv(__filename);
+
+const JobsRowSchema = z.object({
+  authn_user_id: IdSchema.optional(),
+  finish_date: DateFromISOString.optional(),
+  id: IdSchema.optional(),
+  number: z.number().optional(),
+  start_date: DateFromISOString.optional(),
+  status: z.string().optional(),
+});
+
+const CourseRequestRowSchema = z.object({
+  approved_by_name: z.string().nullable(),
+  first_name: z.string().nullable(),
+  github_user: z.string().nullable(),
+  id: IdSchema,
+  institution: z.string().nullable(),
+  jobs: JobsRowSchema,
+  last_name: z.string().nullable(),
+  short_name: z.string().nullable(),
+  status: z.enum(['pending', 'approved', 'denied', 'creating', 'failed']),
+  title: z.string().nullable(),
+  user_id: z.string().nullable(),
+  user_name: z.string().nullable(),
+  work_email: z.string().nullable(),
+});
+
+export async function getCourseRequests(show_all: boolean) {
+  return await queryRow(sql.get_requests, { show_all }, z.array(CourseRequestRowSchema));
+}
```

Comment:
Bikeshedding: calls of this function are currently a little hard to decipher:

```ts
await getCourseRequests(true);
await getCourseRequests(false);
```

It's not clear what this magical boolean is. Consider introducing two function that wrap this one instead:

```ts
selectAllCourseRequests();
selectPendingCourseRequests();
```

This will make it immediately obvious what we're doing at the callsites. At that point, you wouldn't have to export `getCourseRequests` at all.

FWIW, this is the kind of thing that would _technically_ belong in `models/course-request.ts`.

Reply:
That makes sense! I have implemented your two function solution that should make it more clear what these calls are doing. (Also, fixed naming convention of `getCourseRequests` to `selectCourseRequests`)